testsuite/tests/apicast/parameters/policies/custom_policy/conftest.py

```diff
@@ -31,17 +39,57 @@ def policy_settings():
 
 
 @pytest.fixture(scope="module")
-def policy_secret(request, staging_gateway):
+def policy_secret(request, staging_gateway, gateway_kind):
     """
     Create an openshift secrets to use as custom policy based on https://github.com/3scale-qe/apicast-example-policy
     """
-    name = blame(request, "secret")
+    name = blame(request, "secret")[-6:]  # 63 characters limit for referencing secret
     secrets = staging_gateway.openshift.secrets
-    secrets.create(name=name, string_data=custom_policy())
+    labels = {}
+    if gateway_kind.id == SCALE_OPERATOR:
+        labels = {"apimanager.apps.3scale.net/watched-by": "apimanager"}
+    else:
+        labels = {"apicast.apps.3scale.net/watched-by": "apicast"}
+    secrets.create(name=name, string_data=custom_policy(), labels=labels)
     yield name
     del secrets[name]
 
 
+@pytest.fixture(scope="module")
+def changed_secret(staging_gateway, policy_secret):
+    """
+    Change secret with policy to check, if operator reconciles it.
+    """
+    # pylint: disable=protected-access
+    generation = staging_gateway.deployment.get_generation()
+    secrets = staging_gateway.openshift.secrets
+    secret = secrets[policy_secret]
+    example = base64.b64decode(secret._data["example.lua"].encode("ascii")).decode("ascii")
+    example = example.replace("X-Example-Policy-Response", "X-Example-Example-Policy-Response")
+
+    secrets.do_action("set", ["data", "secret/" + policy_secret, f"--from-literal=example.lua={example}"])
+    staging_gateway.reload()
+    staging_gateway.deployment.wait_for()
+
+    def pods_ready(pods):
+        pod_objects = pods.objects()
+        status = bool(pod_objects)
+        for pod in pod_objects:
+            pod.refresh()
+            obj_status = pod.model.status.conditions
+            status = status and all(st["status"] == "True" for st in obj_status)
+        return status
+
+    wait_until = backoff.on_predicate(backoff.fibo, max_tries=30)
+    with ExitStack() as stack:
+        staging_gateway.openshift.prepare_context(stack)
+        pods = staging_gateway.deployment.get_pods()
+        wait_until(lambda: pods_ready(pods))
+    assert generation + 1 == staging_gateway.deployment.get_generation()
```

Comment:
`test_reconcilation[3scale operator]` errors out here, `assert (12 + 1) == 14`

Reply:
@mdujava  is it resolved ?